packages/frontend/src/utils/fetch.ts

```diff
@@ -7,8 +7,9 @@ import type { Payload } from '@hapi/boom';
 type Routes = AMARoutes & AuthRoutes & ModmailRoutes;
 
 export class APIError extends Error {
-	public constructor(public readonly payload: Payload) {
+	public constructor(public readonly payload: Payload, public readonly method: string) {
 		super(payload.message);
+		this.method = method;
```

Comment:
No, adding access modifiers in the constructor (i.e. `public x: T`) already does `this.x = x;` for you.